driver/src/driver/stablex_driver.rs

```diff
@@ -41,8 +41,18 @@ impl<'a> StableXDriver<'a> {
             solution
         };
 
-        let submitted = if solution.executed_sell_amounts.iter().sum::<u128>() > 0 {
-            self.contract.submit_solution(batch, orders, solution)?;
+        let submitted = if solution.is_non_trivial() {
+            let objective_value = self.contract.get_solution_objective_value(
```

Comment:
Maybe it is worth mentioning in a comment that this not just checks the conformity of the solution, but also that the following conditions are met:
// solution is better than the existing solution
// batchId still valid

Reply:
Let me know if the comment I added makes sense